plugins/sql-language/src/main/java/org/polypheny/db/sql/language/SqlIntervalLiteral.java

```diff
@@ -93,20 +93,16 @@ public static class IntervalValue extends PolyValue {
          * Creates an interval value.
          *
          * @param intervalQualifier Interval qualifier
-         * @param sign Sign (+1 or -1)
-         * @param intervalStr Interval string
          */
-        IntervalValue( SqlIntervalQualifier intervalQualifier, int sign, String intervalStr ) {
-            super( null );
-            assert (sign == -1) || (sign == 1);
-            assert intervalQualifier != null;
-            assert intervalStr != null;
+        IntervalValue( SqlIntervalQualifier intervalQualifier, PolyInterval interval ) {
+            super( interval.millis, interval.months );
             this.intervalQualifier = intervalQualifier;
-            this.sign = sign;
-            this.intervalStr = intervalStr;
+            this.sign = interval.millis < 0 ? -1 : 1;
+            this.intervalStr = SqlIntervalQualifier.intervalString( interval, intervalQualifier );
```

Comment:
Just to make sure: Is it correct that the interval stays negative even though there is a separate sign attribute?

Reply:
Is now only used as `isNegative `check, when extracting parts. Replaced and also added months part.